build-pyg-support.py

```diff
@@ -0,0 +1,235 @@
+        os.chdir(args.feedstock_dir)
+
+    def build(self, args):
+        cuda = args.cuda.replace(".", "")
+        if cuda != "none":
+            cuda = "cu" + cuda
+        else:
+            cuda = "cpu"
+
+        run_command([str(self.conda_dir / "build_conda.sh"), args.python, PACKAGES_META["torch"].tag, cuda], run=False)
+
+
+def build_package(package, args):
+    print(f"Building package {package} with pytorch_version={PACKAGES_META['torch'].tag} and cuda_version={args.cuda}")
+    version = None
+    with tempfile.TemporaryDirectory() as tmpdir:
+        feedstock = GitClone(pathlib.Path(tmpdir), package)
+        version = feedstock.get_version()
+        feedstock.apply_patch(args)
+        feedstock.build(args)
+
+    package_path = pathlib.Path.home().absolute() / "conda-bld" / "linux-64"
+
+    cuda = args.cuda.replace(".", "")
+    if cuda == "none":
+        cuda = "cpu"
+
+    tarfile = find_file(
+        package_path,
+        f"{PACKAGES_META[package].name}-{version}-py{args.python.replace('.', '')}"
+        + f"_torch_{PACKAGES_META['torch'].tag}_*{cuda}_openmpi.tar.bz2",
+    )
+    print(f"Package {package} built")
+    return str(package_path / tarfile)
+
+
+def build(args):
+    package_locations = []
+    threw_error = True
+    try:
+        if "torch" in args.package:
+            args.package.remove("torch")
+            package_locations.append(build_pytorch(args))
+
+        should_build_pyg = False
+        if "pyg" in args.package:
+            args.package.remove("pyg")
+            should_build_pyg = True
+
+        for package in args.package:
+            package_locations.append(build_package(package, args))
+
+        if should_build_pyg:
+            package_locations.append(build_package("pyg", args))
+
+        threw_error = False
+    finally:
+        if threw_error:
+            if len(package_locations) == 0:
+                print("No packages built before exiting.")
+            else:
+                print("\nBuild incomplete.\nThe following packages were built before exiting.")
+                for location in package_locations:
+                    print(location)
+                print()
+
+    print("All packages built")
+    with open(args.feedstock_dir / "package_locations.txt", "w") as location_file:
+        for location in package_locations:
+            location_file.write(location + "\n")
+    print(f"Package locations written to '{str(args.feedstock_dir)}/package_locations.txt'")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Builds packages supporting PyG with OpenMPI")
+
+    parser.add_argument(
+        "-p",
+        "--package",
+        type=str,
+        action="append",
+        help="The package(s) to be built",
+        choices=list(PACKAGES_META.keys()),
+        default=None,
+    )
+
+    parser.add_argument(
+        "--python", type=str, help="The version of python used when building packages.", choices=["3.8"], required=True
```

Comment:
I don't think there is a reason to limit the choices here. We will want to move to later python version soon and even build multiple versions, so we will need to update this a lot if we limit it.